src/commands/deployment/correlate.ts

```diff
@@ -76,45 +77,45 @@ export class DeploymentCorrelateCommand extends Command {
       maxConcurrentProcesses: 2, // max 2 git commands at the same time
     })
 
-    const currentBranch = await gitCurrentBranch(git)
-    if (!currentBranch) {
-      this.logger.error('Could not get current branch')
+    if (!this.configurationRepo) {
+      this.configurationRepo = await gitRepositoryURL(git)
+    }
+
+    if (this.configurationRepo === undefined || this.configurationRepo === '') {
+      this.logger.error('Could not retrieve repository URL, check out a repository or provide it with --config-repo')
 
       return 1
     }
 
-    let localCommitShas: string[]
-    if (this.configurationRepo) {
-      localCommitShas = await gitLocalCommitShas(git, currentBranch)
-    } else {
-      ;[this.configurationRepo, localCommitShas] = await Promise.all([
-        gitRepositoryURL(git),
-        gitLocalCommitShas(git, currentBranch),
-      ])
+    if (!this.configurationShas) {
+      const currentBranch = await gitCurrentBranch(git)
```

Comment:
Can we log something here? e.g. Retrieving current commit from Git configuration or something that makes more sense 😬